### PR TITLE
Add Gradle build scripts #44

### PR DESCRIPTION
**Pull request by eerohele**
A couple of suggestions on how to make the build more Gradle-friendly:

- [ ] Move all DITA files into a subdirectory dedicated for DITA files and related images (for example, `src` or `dita` or something like that)
- [ ] Modify the directory structure so that there are no manually maintained and automatically generate DITA files in the same directory (e.g. `parameters`)

The reason for both of these is to make the up-to-date check work better: currently, DITA-OT Gradle Plugin will think the input files have changed and trigger a new build if *any* file under the directory where the input file is is modified, excluding the Gradle build directory and the `.gradle` directory.

(I considered teaching DITA-OT Gradle Plugin to parse DITA files and only consider referenced files in the up-to-date check, but that might have a pretty significant adverse effect on performance.)

If you encounter any issues with the DITA-OT Gradle Plugin, please don't hesitate to [file an issue](https://github.com/eerohele/dita-ot-gradle/issues).

### CODE REVIEW

**Comment by infotexture** [comment]
:+1: Thanks!

Tested `build.gradle` & `site.gradle` from this feature branch on local `docs` clone, both completed successfully. 

Initial run times were 45s for main docs build, 22s for site build. 

After enabling the Gradle [daemon](https://docs.gradle.org/2.8/userguide/gradle_daemon.html), build times dropped dramatically on subsequent builds:

| Build   | `build.gradle` | `site.gradle` |  
| ------- | -------------: | ------------: |  
| Initial |            45s |           22s |  
| Daemon1 |            30s |           24s |  
| Daemon2 |            24s |            8s |  
| Daemon3 |            21s |            1s |  

I also tested the `--continuous` build option provided by the [DITA-OT Gradle Plugin](http://eerohele.github.io/dita-ot-gradle/build/) and verified that builds re-run whenever source files change — which is very convenient for local testing.

I would be inclined to merge this as is, and refactor the directory structure as suggested soon thereafter.

@jelovirt: Any thoughts?

File: site.gradle [hunk 1]
```diff
@@ -0,0 +1,63 @@
+plugins {
```
Comment (jelovirt) [inline site.gradle]: If we have a site.gradle file, we don't need site.xml anymore, right? I think this PR should remove site.xml

Comment (eerohele) [inline site.gradle]: Agreed. Removed in 1642769.

**Comment by jelovirt** [comment]
@eerohele could you squash and update PR?

**Comment by eerohele** [comment]
Done.

**Pull request merged**